clients/highwater/highwater.go

```diff
@@ -68,29 +69,43 @@ func (b *HighwaterClientBuilder) WithVersion(val string) *HighwaterClientBuilder
 	return b
 }
 
+func (b *HighwaterClientBuilder) WithSkip(val bool) *HighwaterClientBuilder {
+	b.config.SkipHighwater = val
+	return b
+}
+
 func (b *HighwaterClientBuilder) WithConfig(val *HighwaterClientConfig) *HighwaterClientBuilder {
-	return b.WithName(val.Name).WithSource(val.MetricsSource).WithVersion(val.MetricsVersion)
+	return b.WithName(val.Name).WithSource(val.MetricsSource).WithVersion(val.MetricsVersion).
+		WithSkip(val.SkipHighwater)
 }
 
 func (b *HighwaterClientBuilder) Build() *HighwaterClient {
+
 	if b.hostGetter == nil {
 		panic("HighwaterClient requires a hostGetter to be set")
 	}
-	if b.config.Name == "" {
-		panic("HighwaterClient requires a name to be set")
-	}
-	if b.config.MetricsSource == "" {
-		panic("HighwaterClient requires a source to be set")
-	}
 
-	if b.config.MetricsVersion == "" {
-		panic("HighwaterClient requires a version to be set")
+	// in case Highwater is skipped these information are not needed
+	if !b.config.SkipHighwater {
+		if b.config.Name == "" {
+			panic("HighwaterClient requires a name to be set")
+		}
+		if b.config.MetricsSource == "" {
+			panic("HighwaterClient requires a source to be set")
+		}
+		if b.config.MetricsVersion == "" {
+			panic("HighwaterClient requires a version to be set")
+		}
```

Comment:
I don't think keeping hostGetter is mandatory ?
```
	// in case Highwater is skipped these information are not needed
	if !b.config.SkipHighwater {
		if b.hostGetter == nil {
			panic("HighwaterClient requires a hostGetter to be set")
		}
		if b.config.Name == "" {
			panic("HighwaterClient requires a name to be set")
		}
		if b.config.MetricsSource == "" {
			panic("HighwaterClient requires a source to be set")
		}
		if b.config.MetricsVersion == "" {
			panic("HighwaterClient requires a version to be set")
		}
	} else {
		log.Printf("Highwater client is configured to skip actually sending metrics to Highwater")
	}
```
wait... the code crash if not here.
Still prettier that way !

Reply:
I think it is needed as Highwater client is not disabled, just not sending actual events to the service